real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/Robot.java

```diff
@@ -0,0 +1,159 @@
+package com.roboclub.robobuggy.main;
+
+import com.roboclub.robobuggy.nodes.localizers.HighTrustGPSLocalizer;
+import com.roboclub.robobuggy.nodes.planners.SweepNode;
+import com.roboclub.robobuggy.nodes.sensors.GpsNode;
+import com.roboclub.robobuggy.nodes.sensors.ImuNode;
+import com.roboclub.robobuggy.nodes.sensors.LoggingNode;
+import com.roboclub.robobuggy.nodes.sensors.RBSMNode;
+import com.roboclub.robobuggy.ros.Node;
+import com.roboclub.robobuggy.ros.NodeChannel;
+
+import java.util.LinkedList;
+import java.util.List;
+
+//TODO break into a separate util class 
+
+/**
+ * 
+ * The user interface controls what gui is  displayed to the user and how they can interact with system 
+ *
+ */
+public final class Robot implements RosMaster {
+
+	/************************************** Sets all internal private variables *************************/
+	private static Robot instance;
+	private boolean autonomous;
+	private boolean collectingData;
+	private static List<Node> nodeList;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Possible unsafe assignment to a non-final static field in a constructor.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2343320404/issues/source?bid=2938665&fileBranchId=3199438#l28)